### PR TITLE
PHP: simplify installation

### PR DESCRIPTION
**Pull request by stanley-cheung**
* For #4517 "PHP installation is too complex" and #4771 "Create binary artifacts for PHP distribution package"
 * Template-ize `config.m4` and `package.xml`. Moved these 2 files all the way up to the base grpc/ directory
 * Now a simple `pear package` command will build the PHP PECL package and gives a package named similar to `grpc-0.8.0.tgz`
 * Now PHP installation can be simplified to these 2 commands (on Debian at least)
   * `sudo apt-get install -y php5 php5-dev php-pear phpunit`
   * `sudo pecl install grpc-beta`
 * Added a bunch of scripts to hook up with Jenkins continuous building and testing the PHP PECL package

### CODE REVIEW

**Comment by stanley-cheung** [comment]
Here's a full Jenkins run building and testing the PHP PECL package: https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_distribtest/118/

File: test/distrib/php/run_distrib_test.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+set -ex
+
+cd $(dirname $0)
+
+cp -r $EXTERNAL_GIT_ROOT/input_artifacts/grpc-php.tgz .
+
+pecl install grpc-php.tgz
+
+php -d extension=grpc.so -d max_execution_time=300 distribtest.php
```
Comment (jtattermusch) [inline test/distrib/php/run_distrib_test.sh]: Would people need to always invoke with php -d extension=grpc.so  when they want to use grpc?

Comment (stanley-cheung) [inline test/distrib/php/run_distrib_test.sh]: Nope. Users can (and should) add that flag to the `php.ini` file. There are various mechanism / location for that file depending on how PHP is invoked (cli, apache, nginx, etc).

File: tools/run_tests/artifact_targets.py [hunk 2]
```diff
@@ -226,6 +226,25 @@ def build_jobspec(self):
                                self.gyp_arch])
 
 
+class PHPExtArtifact:
+  """Builds PHP native extension"""
+
+  def __init__(self, platform, arch):
+    self.name = 'php_ext_{0}_{1}'.format(platform, arch)
```
Comment (jtattermusch) [inline tools/run_tests/artifact_targets.py]: nit: this builds also the pecl  package itself, so the PHPExt name is not really accurate.

**Comment by jtattermusch** [comment]
this is ok to test

**Comment by jtattermusch** [comment]
Besides the minor nits, basically looks good to me.

**Comment by jtattermusch** [comment]
ping, are you planning to fix the nits?

**Comment by stanley-cheung** [comment]
* Fixed the nit
 * Basic tests passed
 * Interop tests failure are unrelated
 * Artifacts tests passed for PHP https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_distribtest/164/

**Pull request merged**